app/src/main/java/org/wikipedia/descriptions/DescriptionEditActivity.java

```diff
@@ -19,26 +20,26 @@
 import org.wikipedia.util.FeedbackUtil;
 import org.wikipedia.util.ShareUtil;
 
+import static org.wikipedia.Constants.INTENT_EXTRA_INVOKE_SOURCE;
+import static org.wikipedia.Constants.InvokeSource;
 import static org.wikipedia.util.DeviceUtil.hideSoftKeyboard;
 
 public class DescriptionEditActivity extends SingleFragmentActivity<DescriptionEditFragment>
         implements DescriptionEditFragment.Callback, LinkPreviewDialog.Callback {
+
     private static final String EXTRA_TITLE = "title";
     private static final String EXTRA_REVIEW_ENABLE = "review";
     private static final String EXTRA_HIGHLIGHT_TEXT = "highlightText";
     private static final String EXTRA_INVOKE_SOURCE = "source";
     private static final String EXTRA_SOURCE_LANG_DESC = "source_desc";
-    public static final int PAGE_SOURCE = 0;
-    public static final int EDIT_TASKS_TITLE_DESC_SOURCE = 1;
-    public static final int EDIT_TASKS_TRANSLATE_TITLE_DESC_SOURCE = 2;
     private ExclusiveBottomSheetPresenter bottomSheetPresenter = new ExclusiveBottomSheetPresenter();
 
     public static Intent newIntent(@NonNull Context context, @NonNull PageTitle title, @Nullable String highlightText, boolean reviewEnabled, int invokeSource, CharSequence sourceDesc) {
         return new Intent(context, DescriptionEditActivity.class)
                 .putExtra(EXTRA_TITLE, GsonMarshaller.marshal(title))
                 .putExtra(EXTRA_HIGHLIGHT_TEXT, highlightText)
                 .putExtra(EXTRA_REVIEW_ENABLE, reviewEnabled)
-                .putExtra(EXTRA_INVOKE_SOURCE, invokeSource)
+                .putExtra(INTENT_EXTRA_INVOKE_SOURCE, invokeSource)
```

Comment:
Do we want to use `putSerializable`?

Reply:
No type info while putting, only getSerializableExtra.